extension/src/popup/views/BuyAsset/MoneyGram/index.tsx

```diff
@@ -1,70 +1,286 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { useTranslation } from "react-i18next";
 import { TextLink } from "@stellar/design-system";
+import StellarSdk, { Account, StellarTomlResolver } from "stellar-sdk";
 
+import { xlmToStroop } from "helpers/stellar";
+import { emitMetric } from "helpers/metrics";
+import {
+  signFreighterTransaction,
+  submitFreighterTransaction,
+  startHwSign,
+  ShowOverlayStatus,
+  transactionSubmissionSelector,
+} from "popup/ducks/transactionSubmission";
+import { LedgerSign } from "popup/components/hardwareConnect/LedgerSign";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+import {
+  publicKeySelector,
+  hardwareWalletTypeSelector,
+} from "popup/ducks/accountServices";
+import { AppDispatch } from "popup/App";
 import { Button } from "popup/basics/buttons/Button";
 import { SubviewHeader } from "popup/components/SubviewHeader";
 import IconGreenCheck from "popup/assets/icon-green-check.svg";
 import MoneyGramLogo from "popup/assets/moneygram-logo.svg";
+import { openTab, navigateTo } from "popup/helpers/navigate";
+import { ROUTES } from "popup/constants/routes";
+import { useNetworkFees } from "popup/helpers/useNetworkFees";
+import { Sep24Status } from "popup/constants/sep24";
+import { METRIC_NAMES } from "popup/constants/metricsNames";
 
 import "./styles.scss";
 
+// TODO - only can buy using test anchor for now
+const testAnchorDomain = "testanchor.stellar.org";
+const testAnchorCode = "SRT";
+const testAnchorIssuer =
+  "GCDNJUBQSX7AJWLJACMJ7I4BC3Z47BQUTMHEICZLE6MU4KQBRYG5JY6B";
+
 export const MoneyGram = () => {
   const { t } = useTranslation();
+  const dispatch: AppDispatch = useDispatch();
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const publicKey = useSelector(publicKeySelector);
+  const [isLoading, setIsLoading] = useState(false);
+  const { recommendedFee } = useNetworkFees();
+  const {
+    hardwareWalletData: { status: hwStatus, lastSignedXDR: hwSignedXDR },
+  } = useSelector(transactionSubmissionSelector);
+  const [freighterSignedXDR, setFreighterSignedXDR] = useState("");
+  const isHardwareWallet = !!useSelector(hardwareWalletTypeSelector);
+  const [sep24Url, setSep24Url] = useState("");
+  const [authEndpoint, setAuthEndpoint] = useState("");
+
+  useEffect(() => {
+    (async () => {
+      const signedXdr = freighterSignedXDR || hwSignedXDR;
+      if (signedXdr) {
+        const token = await getAuthToken(signedXdr);
+        startSep24Deposit(token);
+      }
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
not super happy about needing this. But otherwise all the helper functions below will have to either be defined in the useEffect or each get their own useCallback which gets pretty ugly, so stuck with the eslint comment

Reply:
So this actually meshes with my overall suggestion for this PR, which is to break out some of the logic out of this component into helpers to separate some of the UI concerns from the SEP-24 API concerns. This will make the code a little more modular while also suppressing the exhaustive-deps warning since the functions will be outside of the component and thus won't re-render:

1) `getAuthToken` can just be a standalone function outside of the component as it is stateless

2) I think `startSep24Deposit` should be a redux action and live inside the `transactionSubmission` duck. We'll need to move where polling starts (likely in `/account`) as you suggested, but that means the deposit's `id` and `token` will need to be saved in storage to be fetched later. `startSep24Deposit` can store the `id` and `token` in Background and update the redux store to let UI know that we have a deposit in process. 

Then, when the user reopens the app, it will load storage and check redux to see if it has an `id`/`token` pair in process, and `startPolling` based on that in the `Account` component. Then it removes the `id`/`token` pair on completion

Reply:
This can all be done in a follow up PR btw! Otherwise, this code mostly looks good 👍

Reply:
yup great suggestions. I think that polling mechanism works. I think it might get hairy if there's multiple deposits in process? 🤔 I'll explore/do in the next PR